osquery/tables/system/windows/wmi_tpm_info.cpp

```diff
@@ -0,0 +1,63 @@
+/**
+ * Copyright (c) 2014-present, The osquery authors
+ *
+ * This source code is licensed as defined by the LICENSE file found in the
+ * root directory of this source tree.
+ *
+ * SPDX-License-Identifier: (Apache-2.0 OR GPL-2.0-only)
+ */
+
+#include <sstream>
+
+#include <osquery/core/tables.h>
+#include <osquery/logger/logger.h>
+
+#include <osquery/core/windows/wmi.h>
+
+namespace osquery {
+namespace tables {
+
+QueryData genTpmInfo(QueryContext& context) {
+  QueryData resultsdata;
+  std::stringstream ss;
+  ss << "SELECT * FROM Win32_Tpm";
+
+  BSTR bstr = ::SysAllocString(L"root\\cimv2\\Security\\MicrosoftTpm");

```

Comment:
A call to SysAllocString can fail, so we'll need to check what was returned here.